src/terra_to_portal_posting.py

```diff
@@ -0,0 +1,1122 @@
+                                                              'Number_of_reads_have_multi-mappings': 'n_reads_with_multi_mappings',
+                                                              'Number_of_candidates': 'n_candidates',
+                                                              'Number_of_mappings': 'n_mappings',
+                                                              'Number_of_uni-mappings': 'n_uni_mappings',
+                                                              'Number_of_multi-mappings': 'n_multi_mappings',
+                                                              'Number_of_barcodes_in_whitelist': 'n_barcodes_on_onlist',
+                                                              'Number_of_corrected_barcodes': 'n_corrected_barcodes',
+                                                              'Number_of_output_mappings_(passed_filters)': 'n_output_mappings',
+                                                              'uni-mappings': 'uni_mappings',
+                                                              'multi-mappings': 'multi_mappings',
+                                                              'total': 'total',
+                                                              'percentage_duplicates': 'pct_duplicates',
+                                                              }
+                                             }
+                                }
+                    }
+
+# IGVF file url parsing regex for accession
+IGVF_URL_PATH_REGEX = re.compile(
+    r'\/.*-files\/(IGVF[A-Z0-9]+|TST[A-Z0-9]+)\/@@download')
+
+# GS file path regex
+GS_FILE_PATH_REGEX = re.compile(
+    r'gs://([a-z0-9\-]+)/submissions/final-outputs/([a-z0-9\-]+)/single_cell_pipeline/([a-z0-9\-]+)/.*/.*/([a-z0-9\-]+)/.*')
+
+
+# Define how to return errors
+@dataclasses.dataclass(frozen=True)
+class PostResult:
+    col_header: str
+    accession: str | None
+    error: Exception | None = None
+
+    @classmethod
+    def Failure(cls, col_header, error):
+        return cls(col_header, None, error)
+
+    @classmethod
+    def Success(cls, col_header, accession):
+        return cls(col_header, accession, None)
+
+    def Description(self) -> str:
+        if self.error:
+            return 'POST Failed: ' + str(self.error)
+        return self.accession
+
+
+@dataclasses.dataclass
+class RunResult:
+    analysis_set_acc: str
+    post_results: list[PostResult]
+    finish_time: str
+
+
+def parse_workflow_uuids_from_gs_path(gs_path: str, gs_path_regex: re.Pattern = GS_FILE_PATH_REGEX) -> str:
+    """Parse workflow UUID from a gs path.
+
+    Args:
+        gs_path (str): The gs path to the workflow
+
+    Returns:
+        str: gsbucket-submission-workflow-subworkflow
+    """
+    matches = re.search(gs_path_regex, gs_path)
+    if matches:
+        uuids = matches.groups()
+    return '_'.join(uuids)
+
+
+def get_accessions_from_post_results(post_results: list) -> list:
```

Comment:
why some functions are never used? For example this one: `get_accessions_from_post_results`. Need to clean up